### PR TITLE
Rename Timers "jobs" to simply "timers"

### PR DESCRIPTION
**Pull request by kurtmckee**
### Other

* User timers are now referred to as "timers" rather than as "jobs".

This results in output at the CLI that shows the term "Timer ID" instead of "Job ID", for example.

### CODE REVIEW

File: src/globus_cli/commands/timer/resume.py
Comment (sirosen) [inline src/globus_cli/commands/timer/resume.py:53]: I would (actually _quite strongly_) prefer that we not remove trailing commas in cases like this, which causes `black` to format onto a single line.
Leaving trailing commas -- and in many cases I intentionally add them for this -- makes `black` split arguments over multiple lines. The result is one of the many benefits of `black`: minimal diffs.

If a keyword argument is added here, I should only have to review that addition, not the potential refolding of text caused by extending past the 88 char limit.

Let's please revert this part of the change and any other updates like this which aren't part of renaming.

File: tests/functional/timer/test_job_operations.py
Comment (ada-globus) [inline tests/functional/timer/test_job_operations.py:10]: Nit: I might not change this note until we've updated the APIs, since technically it _is_ a job response today (and hopefully once we make it a proper timer response, we can return and remove the note entirely!)

File: src/globus_cli/commands/timer/_common.py
Comment (sirosen) [inline src/globus_cli/commands/timer/_common.py:79]: I want to voice a general concern I have about making changes like this, although in this case I think it's 99-100% fine.
Users can (and, in the past at least, often did) parse the text output using unix tools. e.g.,
```bash
timer_id="$(globus timer create transfer ... | \
  awk -F':' '$1=="Job ID"{print $2}')" | tr -d ' '
```

There are three reasons I think this particular one is not likely to be an issue:
- this mode of usage was much more common before `--jmespath` was added to the CLI and we started actively advocating that users leverage that flag to parse response data
- Timers is still a relatively young service
- this is the primary object ID, so there is no utility in parsing it out of `globus timer show` output

However, I want us to acknowledge that even though we don't _recommend_ parsing the text output, changing it always carries the risk/possibility that it breaks some downstream user script. I think we should include this in the changelog. It doesn't need to be a separate bullet point -- my thought is to append something like "In text output, the ID of a Timer is now referred to as `Timer ID`."

**Review by sirosen — CHANGES_REQUESTED** [review]

**Review by ada-globus — APPROVED** [review]
I've got a _slight_ uneasiness about a little confusion between jobs and timers from an internal POV (i.e., internal naming) since things like "job_id" are referring to the specific key in the response and/or route they're interacting with. But I am super enthusiastic about the changes to the customer-facing messaging, and am not going to let my very modest reservation hold up what is overall a super positive change!

File: src/globus_cli/commands/timer/_common.py
Comment (kurtmckee) [reply src/globus_cli/commands/timer/_common.py:79]: I can add some mention of that.

Having parsed text output of commands for over a decade, I can tell you that parsing text instead of the JSON output that's readily available is wrong, and we should recommend parsing the JSON directly.

File: src/globus_cli/commands/timer/resume.py
Comment (kurtmckee) [reply src/globus_cli/commands/timer/resume.py:53]: To be fair, I didn't simply remove the trailing comma -- I manually reformatted it myself because it fit on a single line. I didn't realize you added trailing commas specifically to force things onto multiple lines, and I'll try to respect that.

I'll undo this change.

File: tests/functional/timer/test_job_operations.py
Comment (kurtmckee) [reply tests/functional/timer/test_job_operations.py:10]: Is this something you want changed?

File: src/globus_cli/commands/timer/_common.py
Comment (sirosen) [reply src/globus_cli/commands/timer/_common.py:79]: I do not disagree in the slightest. But that has never stopped anyone from writing perl. :stuck_out_tongue_winking_eye:

File: tests/functional/timer/test_job_operations.py
Comment (ada-globus) [reply tests/functional/timer/test_job_operations.py:10]: I didn't request a change b/c I didn't feel strongly enough (or that the impact was significant enough) to insist on it, only that it registered on my radar. I leave it to your judgment, but will not be put out if you leave it as is.

File: src/globus_cli/commands/timer/resume.py
Comment (kurtmckee) [reply src/globus_cli/commands/timer/resume.py]: This should be undone (simply kicked the change out of the previous commit and force-pushed).

File: src/globus_cli/commands/timer/_common.py
Comment (kurtmckee) [reply src/globus_cli/commands/timer/_common.py:79]: I've updated the changelog fragment to give an example that output shows "Timer ID" instead of "Job ID".

**Review by sirosen — APPROVED** [review]

**Pull request merged**
